src/Lecture1.hs

```diff
@@ -121,21 +119,8 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-split :: Char -> String -> String -> [String]
-split c [] [] = []
-split c acc [] = [acc]
-split c acc (x:xs)
-    | (x == c) && (not $ null acc) = [acc] ++ (split c "" xs)
-    | x == c = split c acc xs
-    | otherwise = split c (acc ++ [x]) xs
-
-sumStr :: [String] -> Int
-sumStr [] = 0
-sumStr (x:xs) = (read x :: Int) + sumStr xs
-
 strSum :: String -> Int
-strSum "" = 0
-strSum str = sumStr $ split ' ' "" str
+strSum str = sum $ map (\x -> read x :: Int) $ words str
```

Comment:
As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body:

```suggestion
strSum str = sum (map read (words str))
```

Reply:
Wow, that looks fantastic. I was struggling with how to combine `read x :: Int` with the map function. 😆